src/options/options_public_template.cpp

```diff
@@ -30,200 +30,166 @@
 #include <limits>
 
 namespace cvc5::options {
-
-bool getUfHo(const Options& opts) { return opts.uf.ufHo; }
-
-/** Set a given Options* as "current" just for a particular scope. */
-class OptionsGuard {
-  Options** d_field;
-  Options* d_old;
-public:
-  OptionsGuard(Options** field, Options* opts) :
-    d_field(field),
-    d_old(*field) {
-    *field = opts;
-  }
-  ~OptionsGuard() {
-    *d_field = d_old;
+  // Contains the default option handlers (i.e. parsers)
+  namespace handlers {
+
+  /**
+   * Utility function for handling numeric options. Takes care of checking for
+   * unsignedness, parsing and handling parsing exceptions. Expects `conv` to be
+   * a conversion function like `std::stod`, accepting a `std::string` and a
+   * `size_t*`. The argument `type` is only used to generate proper error
+   * messages and should be the string representation of `T`. If `T` is
+   * unsigned, checks that `optionarg` contains no minus. Then `conv` is called
+   * and the error conditions are handled: `conv` may throw an exception or
+   * `pos` may be smaller than the size of `optionarg`, indicating that not the
+   * entirety of `optionarg` was parsed.
+   */
+  template <typename T, typename FF>
+  T parseNumber(const std::string& flag,
+                const std::string& optionarg,
+                FF&& conv,
+                const std::string& type)
```

Comment:
Is it possible to derive the name of the type as a string from T?

Reply:
We could do `typeid(T).name()`, but for example `double` would be printed as `d`...

Reply:
Ok, then let's keep it the way it is.